core.network/src/org/netbeans/core/network/utils/hname/win/HostnameUtilsWin.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.core.network.utils.hname.win;
+
+import com.sun.jna.Native;
+import com.sun.jna.platform.win32.Kernel32Util;
+import com.sun.jna.platform.win32.Win32Exception;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.netbeans.core.network.utils.IpAddressUtils;
+import org.netbeans.core.network.utils.NativeException;
+
+/**
+ * Hostname utilities for Microsoft Windows OS.
+ */
+public class HostnameUtilsWin {
+    
+    private static final Logger LOGGER = Logger.getLogger(HostnameUtilsWin.class.getName());
+    
+    /**
+     * Gets the computer name.
+     * 
+     * <p>This is the also known as the NetBIOS name, although NetBIOS is 
+     * hardly used anymore. It is the same value as can be seen from the
+     * {@code COMPUTERNAME} environment variable.
+     * 
+     * <p>
+     * Windows API equivalent: {@code GetComputerName()} function from 
+     * {@code Kernel32} library.
+     * 
+     * @return computer name
+     * @throws NativeException if there was an error executing the
+     *    system call.
+     */
+    public static String getComputerName() throws NativeException {
```

Comment:
This function is not used and I would remove this and the logger above. When this is done, the dependency on JNA Platform should also be removed. I noticed this, because I thought about moving the JNA function definitions for the function into core.network. JNA Platform weights > 2 MB and if we can get away without it, I think this would be good.

As you don't need it, please remove.

Reply:
True, that it is not used in *my* implementation. I used to use it, but then I switched to using `getHostName` instead.  There are pros and cons on these two Win32 methods and someone implementing differently than me may feel differently about it. So I kept it.

The dependency on JNA Platform will come back with a vengeance in the next iterations on the module where I improve further on the module. So no point in removing it now. Remember, as it says in the top text: The improved PAC script evaluator is just the beginning. To give you a taste: The next steps will for example add vastly improved config discovery and for this JNA Platform is needed.

Reply:
About size: If that is a concern, how about the JNA project would consider splitting Platform jar into one jar per OS ?   :-)  COM in its own jar, of course, in order to further reduce size.

Anyways, I don't see a size of > 2MB. When I look in the `.nbm` file I see a pack200'ed jar of approx 178 KB. 

But if you feel it is important to get rid of JNA Platform lib then I'll be happy to extract the bits I need into the source. And I'll have to do the same in the next iterations of `core.network`, of course.